src/main/java/de/tum/in/www1/artemis/domain/competency/CompetencyJol.java

```diff
@@ -23,15 +25,18 @@
 public class CompetencyJol extends DomainObject {
 
     @ManyToOne
-    @JoinColumn(name = "competency_id")
+    @JoinColumn(name = "competency_id", nullable = false)
     private Competency competency;
 
     @ManyToOne
-    @JoinColumn(name = "user_id")
+    @JoinColumn(name = "user_id", nullable = false)
     private User user;
 
     @Column(name = "jol_value")
-    private Integer value;
+    private short value = 0;
+
```

Comment:
How can I know that the mastery changed? I think this is missing the `mastery` as a column.